File: Question 0024 - Swap Nodes in Pairs/src/Solution.java
import java.util.Stack;

class Solution {
    public ListNode swapPairs(ListNode head)
    {
        ListNode dummy = new ListNode(-1);
        ListNode dummyPtr = dummy;
        Stack<ListNode> st = new Stack<>();

        ListNode ptr = head;

        while(ptr != null)
        {
            ListNode ptrNext = ptr.next;
            for (int i = 0; i < 2 && ptr!= null; i++) {
                st.push(ptr);
                ptr = ptrNext;
                if(ptrNext != null) ptrNext = ptrNext.next;
            }

            while(!st.isEmpty())
            {
                dummyPtr.next = st.pop();
                dummyPtr = dummyPtr.next;
            }

        }

        dummyPtr.next = null;
        return dummy.next;
    }
}

File: Question 0024 - Swap Nodes in Pairs/src/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution 
{
public:
    ListNode* swapPairs(ListNode* head) 
    {
        ListNode ans(-1);
        ListNode* ansPtr = &ans;
        stack<ListNode*> st;
        ListNode* headPtr = head;

        while(headPtr != NULL)
        {
            for(int i = 0; i < 2 && headPtr != NULL; i++)
            {
                st.push(headPtr);
                headPtr = headPtr->next;
            }

            while(!st.empty())
            {
                ansPtr->next = st.top();
                st.pop();
                ansPtr = ansPtr->next;
            }
        }    

        ansPtr->next = NULL;
        return ans.next;
    }
};